engine/src/main/java/org/pentaho/di/trans/step/StepMetaInterface.java

```diff
@@ -722,6 +722,15 @@ default boolean cleanAfterHopFromRemove() {
     return false;
   }
 
+  /**
+   * Action remove hop from this step
+   * @param toStepName - String of step name of the "to" step of the hop
+   * @return step was changed
+   */
+  default boolean cleanAfterHopFromRemove( String toStepName ) {
```

Comment:
Do we still need the no-arg version of this method? If not, and it's too dangerous to remove it at this point, would it make sense to mark it deprecated?

Reply:
not quite sure if anything else uses it...non-kettle related. doesn't hurt to leave it if it's not used

Reply:
I searched all pentaho projects on github and didn't find any references. I understand removing it might be risky, just wanted to point it out. If you're not sure, it's fine to leave it.